packages/material-ui-lab/src/Masonry/Masonry.js

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import {
+  createUnarySpacing,
+  getValue,
+  handleBreakpoints,
+  unstable_resolveBreakpointValues as resolveBreakpointValues,
+} from '@material-ui/system';
+import { deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import { styled, useThemeProps } from '@material-ui/core/styles';
+import { getMasonryUtilityClass } from './masonryClasses';
+import MasonryContext from './MasonryContext';
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getMasonryUtilityClass, classes);
+};
+
+export const style = ({ styleProps, theme }) => {
+  let styles = {
+    display: 'grid',
+    gridAutoRows: 0,
+    padding: 0,
+    overflow: 'auto',
+    width: '100%',
+    rowGap: 2,
+    boxSizing: 'border-box',
+  };
+
+  const base = {};
+  Object.keys(theme.breakpoints.values).forEach((breakpoint) => {
+    if (styleProps.spacing[breakpoint] != null) {
+      base[breakpoint] = true;
+    }
+  });
+
+  const spacingValues = resolveBreakpointValues({ values: styleProps.spacing, base });
+  const transformer = createUnarySpacing(theme);
+  const spacingStyleFromPropValue = (propValue) => {
+    return {
+      columnGap: getValue(transformer, propValue),
+    };
+  };
+
+  styles = {
+    ...styles,
+    ...handleBreakpoints({ theme }, spacingValues, spacingStyleFromPropValue),
+  };
+
+  const columnValues = resolveBreakpointValues({ values: styleProps.columns, base });
+  const columnStyleFromPropValue = (propValue) => {
+    return {
+      gridTemplateColumns: `repeat(${propValue}, 1fr)`,
+    };
+  };
+
+  styles = deepmerge(styles, handleBreakpoints({ theme }, columnValues, columnStyleFromPropValue));
+
+  return styles;
+};
+
+const MasonryRoot = styled('div', {
+  name: 'MuiMasonry',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    return [styles.root];
+  },
+})(style);
+
+const Masonry = React.forwardRef(function Masonry(inProps, ref) {
+  const props = useThemeProps({
+    props: inProps,
+    name: 'MuiMasonry',
+  });
+
+  const { children, className, component = 'div', columns = 4, spacing = 1, ...other } = props;
+  const styleProps = { ...props, spacing, columns };
```

Comment:
```suggestion
  const ownerState = { ...props, spacing, columns };
```

Let's rename it everywhere to be consistent with the other components.